addons/hr_recruitment/migrations/9.0.1.0/post-migration.py

```diff
@@ -27,22 +27,18 @@ def update_applicant_availability(cr):
 
 
 def migrate_applicant_source(cr):
-    cr.execute(
-        """
-            SELECT id
-            FROM hr_recruitment_source
-        """)
-    for old_id in cr.fetchall():
-        cr.execute("INSERT INTO utm_source (name) "
-                   "SELECT name "
-                   "FROM %s "
-                   "RETURNING id",
-                   (openupgrade.get_legacy_name('hr_recruitment_source'),))
-        new_id = cr.fetchone()[0]
-        cr.execute("UPDATE hr_applicant "
-                   "SET source_id = %s "
-                   "WHERE source_id = %s",
-                   (new_id, old_id[0]))
+    env = api.Environment(cr, SUPERUSER_ID, {})
```

Comment:
Use this as faster operation and usual way to do things here in OpenUpgrade:
```python
column = openupgrade.get_legacy_name('hr_recruitment_source')
openupgrade.logged_query(cr, "ALTER TABLE utm_source ADD %s INT4", (column, ))
openupgrade.logged_query(
    cr, """
    INSERT INTO utm_source
    (create_uid, create_date, write_uid, write_date, name, %s)
    SELECT create_uid, create_date, write_uid, write_date, name, id
    FROM hr_recruitment_source
    """, (column, ),
)
openupgrade.logged_query(
    cr, """
    UPDATE hr_applicant ha SET source_id = us.id
    FROM utm_source us WHERE us.%s = ha.%s
    """, (column, openupgrade.get_legacy_name('source_id')),
)
```
Maybe for avoiding flake8 you need to put `AsIs` in the names of fields.

Reply:
The number of recruitement sources is likely to be limited in almost any case isn't it? Using the ORM seems fine to me.

Reply:
But then the previous code was OK except the renaming of the column. No need to make a rewriting of something that is not better.

Reply:
The perfect is the enemy of the good. The new code is not worse than the old code. In fact it's better because it contains the bugfix. So I'm asking you to reconsider so we can merge this fix.

Reply:
I have given the needed code, so in this case I prefer to have the "perfect" solution. If not, I can do the PR with my proposal.

Reply:
That's the true optimistic merging style that I still favour! Merge quickly, and create a follow-up PR to fix what could be improved. Proceeding with the merge

/ocabot merge